src/__tests__/lib/utils/functions/EntityTypeUtils.test.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  convertToEntityType,
+  entityTypeToFriendlyName,
+  isContainerType,
+  isVersionableEntityType,
+} from '../../../../lib/utils/functions/EntityTypeUtils'
+import { EntityType } from '../../../../lib/utils/synapseTypes'
+
+const allEntityTypes = [
+  {
+    type: EntityType.PROJECT,
+    concreteType: 'org.sagebionetworks.repo.model.Project',
+    friendlyName: 'Project',
+  },
+  {
+    type: EntityType.FOLDER,
+    concreteType: 'org.sagebionetworks.repo.model.Folder',
+    friendlyName: 'Folder',
+  },
+  {
+    type: EntityType.LINK,
+    concreteType: 'org.sagebionetworks.repo.model.Link',
+    friendlyName: 'Link',
+  },
+  {
+    type: EntityType.DOCKER_REPO,
+    concreteType: 'org.sagebionetworks.repo.model.docker.DockerRepository',
+    friendlyName: 'Docker Repository',
+  },
+  {
+    type: EntityType.FILE,
+    concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+    friendlyName: 'File',
+  },
+  {
+    type: EntityType.TABLE,
+    concreteType: 'org.sagebionetworks.repo.model.table.TableEntity',
+    friendlyName: 'Table',
+  },
+  {
+    type: EntityType.SUBMISSION_VIEW,
+    concreteType: 'org.sagebionetworks.repo.model.table.SubmissionView',
+    friendlyName: 'Submission View',
+  },
+  {
+    type: EntityType.ENTITY_VIEW,
+    concreteType: 'org.sagebionetworks.repo.model.table.EntityView',
+    friendlyName: 'View',
+  },
+  {
+    type: EntityType.DATASET,
+    concreteType: 'org.sagebionetworks.repo.model.table.Dataset',
+    friendlyName: 'Dataset',
+  },
+  {
+    type: EntityType.MATERIALIZED_VIEW,
+    concreteType: 'org.sagebionetworks.repo.model.table.MaterializedView',
+    friendlyName: 'Materialized View',
+  },
+]
+
+/**
+ * The purpose of these tests is more to ensure that we're handling all entity types as new ones are added.
+ * A key assumption is that that the EntityType enum is the "source of truth", so these tests will check that we handle all of the entity types in our test list.
+ */
```

Comment:
IOW, I really only care that we will see tests fail when we add a new entity type.